core/src/main/java/org/apache/accumulo/core/client/impl/MasterClient.java

```diff
@@ -41,12 +41,19 @@
 public class MasterClient {
   private static final Logger log = LoggerFactory.getLogger(MasterClient.class);
 
-  public static MasterClientService.Client getConnectionWithRetry(ClientContext context) {
+  public static MasterClientService.Client getConnectionWithRetry(ClientContext context, int numberOfRetries) {
```

Comment:
Instead of passing a count, could pass a max time.   I think passing a time assumes less about the implementation.  Could do something like the following.

```java
public static MasterClientService.Client getConnectionWithRetry(ClientContext context, long maxRetryTime, TimeUnit retryTimeUnit) {

```